### ISSUE TITLE
Crash NPE on onVenmoFailure listener is null

### ISSUE THREAD

**Issue by bruno-facina-penn-int** [braintree/braintree_android#832]
### Braintree SDK Version

4.33.0

### Environment

Production

### Braintree dependencies

com.braintreepayments.api:paypal:4.33.0,
com.braintreepayments.api:venmo:4.33.0,
com.braintreepayments.api:data-collector:4.33.0,

### Describe the bug

`NullPointerException - Attempt to invoke interface method 'void com.braintreepayments.api.i2.a` caused by `VenmoClient.java:305`
The listeners are null at that point, oddly it only happens for failure, it is very rare occasions, something is happening with the activity that we are losing the listeners, we also have set the listeners for paypal the same way, but we have no issue there. When looking into your codebase, we have found that paypal checks for the listener to not be null

PaypalClient.java:155
```
if (error != null && listener != null) {
                    listener.onPayPalFailure(error);
}
```

while for Venmo there are no checks for the listener not being null,
VenmoClient.java:305

```
else if (venmoResult.getError() != null) {
            if (venmoResult.getError() instanceof UserCanceledException) {
                braintreeClient.sendAnalyticsEvent("pay-with-venmo.app-switch.canceled");
            }
            listener.onVenmoFailure(venmoResult.getError());
        }
```

### To reproduce

We couldn't reproduce, it has been happening in production.

### Expected behavior

The fix for this is very simple, only call the listeners for the VenmoClient if they are not null.

**Comment by hollabaq86** [comment]
for internal tracking, issue 30167

**Comment by sarahkoop** [comment]
This fix was released in version `4.40.1`
